projects/mathjax-lib/README.md

```diff
@@ -1,24 +1,144 @@
-# MathjaxLib
+# mathjax-angular 
+This plugins implements the browser version of [Mathjax v3][1] into [Angular][2]
+## Feature
 
-This library was generated with [Angular CLI](https://github.com/angular/angular-cli) version 12.1.1.
+- Dynamically load MathJax library to your web application.
+- Simple typesetting using Angular directive.
+- Dynamic typesetting using expressions.
 
-## Code scaffolding
+## Install
 
-Run `ng generate component component-name --project mathjax-lib` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module --project mathjax-lib`.
-> Note: Don't forget to add `--project mathjax-lib` or else it will be added to the default project in your `angular.json` file. 
+NPM:
+```
+npm install mathjax-angular
+```
 
-## Build
+Yarn:
+```
+yarn add mathjax-angular
+```
 
-Run `ng build mathjax-lib` to build the project. The build artifacts will be stored in the `dist/` directory.
+## Configure the module
 
-## Publishing
+Load the module in the `@NgModule` class of the application. You need
+to pass a `ModuleConfiguration` instance to the `config` method to
+configure the module.
 
-After building your library with `ng build mathjax-lib`, go to the dist folder `cd dist/mathjax-lib` and run `npm publish`.
+### Example
 
-## Running unit tests
+We provide a premade exaple with the library itself, You can see the example [here][3], or alternatively run it locally by cloning the [repository][4] and running:
+```
+npm install&&npm run dev
+```
 
-Run `ng test mathjax-lib` to execute the unit tests via [Karma](https://karma-runner.github.io).
+### Usage
+When importing in the **root** module, the module should be configured
+with *.forRoot* method.
 
-## Further help
+```typescript
+import {NgModule} from '@angular/core';
+import {AppComponent} from './src/app/app.component';
+import {MathJaxModule} from './src/app/math-jax/math-jax.module';
 
-To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI Overview and Command Reference](https://angular.io/cli) page.
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    MathJaxModule.forRoot() //Pre configured
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+When importing in a **child** module, the module must be configured to
+re-use the same module instance as the root module. So simply
+configure the module with the *.forChild* method.
+
+```typescript
+import {MathJaxModule} from './src/app/math-jax/math-jax.module';
+
+...
+imports: [
+  MathJaxModule.forChild()
+]
+...
+```
+
+## Typeset an element
+
+Add the `mathjax` directive to elements which you want to apply
+MathJax typesetting on load.
+
+```html
+<div>normal text</div>
+
+<div mathjax>mathjax typesetting
```

Comment:
remove this
in our implementation we need to pass mathjax value as attribute

Reply:
`<div [mathjax]="content"></div>
// in your ts component
content = `$x = {-b \\pm \\sqrt{b^2-4ac} \\over 2a}$`
`

Reply:
## Typeset an element

Add the `mathjax` directive to elements which you want to apply
MathJax typesetting on load.

```html
<div [mathjax]="content"></div> 
// in your ts component content = "$x = {-b \\pm \\sqrt{b^2-4ac} \\over 2a}$"
```